src/dev.c

```diff
@@ -158,16 +158,14 @@ fido_dev_open_tx(fido_dev_t *dev, const char *path)
 	    dev->report_in_len > CTAP_MAX_REPORT_LEN) {
 		fido_log_debug("%s: invalid report_in_len %hu", __func__,
 		    dev->report_in_len);
-		/* Fall back to default USB transfer report length. */
-		dev->report_in_len = CTAP_MAX_REPORT_LEN;
+		return (FIDO_ERR_INTERNAL);
```

Comment:
Makes sense. Given we have opened the device and assigned dev->io_handle at that point, we need to

```c
dev->io.close(dev->io_handle);
dev->io_handle = NULL;
```

before returning.

Reply:
Thanks! I fixed that and also changed the returned error code to distinguish between issues with the in or out report - I found `FIDO_ERR_TX` and `FIDO_ERR_RX` appropriate since these would have been the ultimately returned error codes before we made libfido2 report size aware.